src/vulkan-renderer/application.cpp

```diff
@@ -583,6 +598,29 @@ void Application::run() {
         process_mouse_input();
         m_camera->update(m_time_passed);
         m_time_passed = m_stopwatch.time_step();
+
+        // Check for collision between camera ray and every octree
```

Comment:
I would move the collision_test block to a method

Reply:
* [ ] Move collision check in main loop into a separate method.

Reply:
Done.